lib/services/internalbroker.js

```diff
@@ -40,7 +40,7 @@ module.exports = function (kuzzle, options) {
    * @return Promise when initialization is complete
    */
   this.init = function () {
-    return (this.isServer) ? serverStart.call(this) : clientConnect.call(this);
+    return (this.isServer) ? q(serverStart.call(this)) : q(clientConnect.call(this));
```

Comment:
Unnecessary: both these functions already return promises, resolved either when the ipc server is started or when the client is connected to a kuzzle server